src/seedu/addressbook/data/person/Address.java

```diff
@@ -36,6 +47,49 @@ public static boolean isValidAddress(String test) {
         return test.matches(ADDRESS_VALIDATION_REGEX);
     }
 
+    /**
+     * Returns an ArrayList of the respective data from an Address
+     */
+    public static ArrayList<String> splitIntoAddressData(String rawAddress) {
+        ArrayList<String> addressData = new ArrayList<String>();
+        int j = 0;
+        int indexOfComma = rawAddress.indexOf(',');
+        while (indexOfComma != -1) {
+            addressData.add(rawAddress.substring(0, indexOfComma));
+            j = indexOfComma + 2;
+            rawAddress = rawAddress.substring(j);
+            indexOfComma = rawAddress.indexOf(',');
+        }
+        addressData.add(rawAddress);
+
+        return addressData;
+    }
+
+    /**
+     * Stores the address data in its respective object attributes
+     */
+    public void initAddressProperties(ArrayList<String> addressData) {
```

Comment:
Use the split() method to make the code shorter.
Assign ', ' to a regex variable.